full-stack-tests/core/src/frontend/standalone/SchemaLocator.test.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelConnection } from "@itwin/core-frontend";
+import { TestUtility } from "../TestUtility";
+import { SchemaContext } from "@itwin/ecschema-metadata";
+import { ECSchemaRpcLocater } from "@itwin/ecschema-rpcinterface-common";
+import * as EC from "@itwin/ecschema-metadata";
+import { assert, expect } from "chai";
+import { TestSnapshotConnection } from "../TestSnapshotConnection";
+
+describe("Schema Locater tests: ", () => {
+  let context = new SchemaContext();
+  let imodel: IModelConnection;
+
+  beforeEach(async () => {
+    await TestUtility.startFrontend();
+    imodel = await TestSnapshotConnection.openFile("testImodel.bim"); // relative path resolved by BackendTestAssetResolver
+    const schemaLocater = new ECSchemaRpcLocater(imodel);
+    context = new SchemaContext();
+    context.addLocater(schemaLocater);
+  });
+
+  afterEach(async () => {
+    if (undefined !== imodel)
+      await imodel.close();
+
+    await TestUtility.shutdownFrontend();
+  });
+
+  it("locate valid schema Async", async () => {
+    const schemaKey = new EC.SchemaKey("Gist", 1, 0, 0);
+    const schema = await context.getSchema(schemaKey, EC.SchemaMatchType.Exact);
+
+    assert.isDefined(schema);
+    assert.strictEqual(schema!.schemaKey.name, "Gist");
+    assert.strictEqual(schema!.schemaKey.version.toString(), "01.00.00");
```

Comment:
I suggest a few more tests:
After calling getSchema() we should do a getSchemaSync, which due to caching is probably successful. This behavior seems inconsistent to me, but at the very least we should cover it with a test so we have it covered.

And another test that uses getSchemaItem/getSchemaItemSync on the context, instead of getSchema.

Reply:
Added new test cases